v2/thirdpartyemailpassword/migration/account-creation/ep-migration-without-password-hash.mdx

```diff
@@ -0,0 +1,232 @@
+                                break;
+                            }
+                        }
+
+                        if (emailPasswordUser === undefined) {
+                            // EmailPassword user does not exist in SuperTokens 
+                            // Validate user credentials in the external provider
+                            let legacyUserInfo = await validateAndGetUserInfoFromExternalProvider(input.email, input.password)
+
+                            if (legacyUserInfo === undefined) {
+                                // Credentials are incorrect in the external provider
+                                return {
+                                    status: "WRONG_CREDENTIALS_ERROR"
+                                }
+                            }
+
+                            // Call the signup function to create a new SuperTokens user.
+                            let response = await ThirdPartyEmailPassword.emailPasswordSignUp(input.email, input.password, input.userContext);
+
+                            if (response.status !== "OK") {
+                                throw new Error("Should never come here");
+                            }
+
+                            // Map the legacy userId to the SuperTokens userId
+                            await SuperTokens.createUserIdMapping({ superTokensUserId: response.user.id, externalUserId: legacyUserInfo.user_id })
+
+                            // Set the userId in the response to use the external userId.
+                            response.user.id = legacyUserInfo.user_id;
+
+                            // We will also need to set the email verification status of the user
+                            if (legacyUserInfo.isEmailVerified) {
+                                // generate an email verification token for the user
+                                let generateEmailVerificationTokenResponse = await EmailVerification.createEmailVerificationToken(response.user.id, input.email, input.userContext);
+
+                                if (generateEmailVerificationTokenResponse.status === "OK") {
+                                    // verify the user's email
+                                    await EmailVerification.verifyEmailUsingToken(generateEmailVerificationTokenResponse.token, input.userContext);
+                                }
+                            }
+                            return response;
+                        }
+                    }
+                    // user does not exist in the external provider so we call the original implementation
+                    return originalImplementation.emailPasswordSignIn(input)
+                },
+            }
+        },
+    }
+})
+
+async function doesUserExistInExternalProvider(email: string): Promise<boolean> {
+    // TODO: check if user with the input email exists in the external provider
+    return false;
+}
+
```

Comment:
need to take into account the flow is the user signs in with correct legacy credentials after generating a password token.